### PR TITLE
Offline payloads

### PR DESCRIPTION
**Pull request by polymath-eric**
### JIRA Link 

[DA-945](https://polymesh.atlassian.net/browse/DA-945)

### Changelog / Description 

Allows for transactions to be signed offline

### Checklist - 

- [X] New Feature ?
- [X] Updated swagger annotation (if API structure is changed) ?
- [X] Unit Test (if possible) ?
- [X] Updated the Readme.md (if required) ?

[DA-945]: https://polymesh.atlassian.net/browse/DA-945?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/common/dto/transaction-options.dto.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    enum: ProcessMode,
+  })
+  @IsEnum(ProcessMode)
+  readonly processMode: ProcessMode;
+
+  // Hide the property so the interactive examples work without additional setup
+  @ApiHideProperty()
+  @ValidateIf(({ processMode }) => processMode === ProcessMode.SubmitWithCallback)
+  @IsString()
+  @IsUrl()
```
Comment (prashantasdeveloper) [inline src/common/dto/transaction-options.dto.ts:27]: `@IsOptional`?

File: src/common/models/transaction-payload.model.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { TransactionPayload } from '@polymeshassociation/polymesh-sdk/types';
+import { Type } from 'class-transformer';
+
+import { TransactionDetailsModel } from '~/common/models/transaction-details.model';
+
+export class TransactionPayloadModel {
+  @ApiProperty({
+    description: 'SDK payload',
+  })
+  unsignedTransaction: TransactionPayload;
```
Comment (prashantasdeveloper) [inline src/common/models/transaction-payload.model.ts:13]:
```suggestion
  readonly unsignedTransaction: TransactionPayload;
```

File: src/common/models/transaction-payload.model.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { ApiProperty } from '@nestjs/swagger';
+import { TransactionPayload } from '@polymeshassociation/polymesh-sdk/types';
+import { Type } from 'class-transformer';
+
+import { TransactionDetailsModel } from '~/common/models/transaction-details.model';
+
+export class TransactionPayloadModel {
+  @ApiProperty({
+    description: 'SDK payload',
+  })
```
Comment (prashantasdeveloper) [inline src/common/models/transaction-payload.model.ts:12]:
```suggestion
  @ApiProperty({
    description: 'SDK payload',
    type: TransactionPayload
  })
  @Type(() => TransactionPayload)
```

File: src/common/models/transaction-payload.model.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  unsignedTransaction: TransactionPayload;
+
+  @ApiProperty({
+    description: 'Transaction details',
+    isArray: true,
+  })
```
Comment (prashantasdeveloper) [inline src/common/models/transaction-payload.model.ts:18]:
```suggestion
  @ApiProperty({
    description: 'Transaction details',
    type: TransactionDetailsModel,
  })
```

Comment (prashantasdeveloper) [reply src/common/models/transaction-payload.model.ts:18]: Parameter is not of array type

File: src/common/models/transaction-payload.model.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  @Type(() => TransactionDetailsModel)
+  details: TransactionDetailsModel;
```
Comment (prashantasdeveloper) [inline src/common/models/transaction-payload.model.ts:20]:
```suggestion
  readonly details: TransactionDetailsModel;
```

File: src/transactions/dto/payload.dto.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import { ApiProperty } from '@nestjs/swagger';
```
Comment (prashantasdeveloper) [inline src/transactions/dto/payload.dto.ts:1]: istanbul ignore

File: src/transactions/dto/payload.dto.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  @Length(66)
+  readonly blockHash: string;
+
+  @ApiProperty({
+    type: 'string',
+    description:
+      'The latest block number when this transaction was created. Used to control transaction lifetime (Alternative to block hash)',
+    example: '0x00000000',
+  })
+  @IsHexadecimal()
```
Comment (prashantasdeveloper) [inline src/transactions/dto/payload.dto.ts:45]: Is this hex as well? Feels a bit weird

File: src/transactions/dto/raw-payload.dto.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { ApiProperty } from '@nestjs/swagger';
```
Comment (prashantasdeveloper) [inline src/transactions/dto/raw-payload.dto.ts:1]: istanbul ignore

File: src/transactions/dto/transaction.dto.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  @Type(() => PayloadDto)
+  @ValidateNested()
+  readonly payload: PayloadDto;
+
+  @ApiProperty({
+    type: RawPayloadDto,
+    description: 'The raw transaction payload',
+  })
+  @Type(() => RawPayloadDto)
+  @IsOptional()
```
Comment (prashantasdeveloper) [inline src/transactions/dto/transaction.dto.ts:42]:
```suggestion

```

File: src/common/dto/transaction-options.dto.ts (hunk 1 above)
Comment (polymath-eric) [reply src/common/dto/transaction-options.dto.ts:27]: If someone uses callback, then they have to provide a webhook URL. I'll add a note

**Comment by polymath-eric** [comment]
merge-blocked

I am going to force push since my merge commit is blocking me

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**